worker/echo/main.go

```diff
@@ -36,8 +43,23 @@ func echo(
 
 	// Sleep time between receiving the message and sending it
 	if sleepTime > 0 {
-		log.Infof("sleeping: %v", sleepTime)
+		// Setting the channel to handle cancelation
+		sycMapCancelChan.Set(rcvId, make(chan struct{}))
+
+		log.Tracef("sleeping: %v", sleepTime)
 		time.Sleep(sleepTime)
+
+		// Cancel message if it has been sent a cancel message
+		// during sleep time
+		cancelChan, _ := sycMapCancelChan.Get(rcvId)
+		select {
+		case <-cancelChan:
+			log.Tracef("canceled echo message id: %v", rcvId)
+			sycMapCancelChan.Del(rcvId)
+			return nil
+		default:
+			sycMapCancelChan.Del(rcvId)
+		}
 	}
```

Comment:
This example feels a bit pedantic to me. Maybe instead of sleeping, the echo
worker could repeatedly echo its message a specific number of times before
"completing" its work. This would allow us to hook a cancel message check into
the work loop, giving a more authentic demonstration of the cancellation.